### PR TITLE
add docs

### PR DESCRIPTION
**Pull request by chpio**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+//! The function is not injective. This means several *UUID*s will give the same name. We don't
+//! consider it an issue since context (like owner of object) will help dedup the search.
+//!
+//! ## Examples
+//! ```rust
+//! extern crate uuid;
+//! extern crate uuid_to_pokemon;
+//!
+//! use uuid::Uuid;
+//! use uuid_to_pokemon::{uuid_to_pokemon, PokemonUuid};
```
Comment (Keruspe) [inline src/lib.rs]: Do we still need to use PokemonUuid here?

File: src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+//! use std::fmt::Write; // needed for write!
+//!
+//! fn main() {
+//!     let my_uuid: Uuid = Uuid::nil();
+//!     let result: PokemonUuid = uuid_to_pokemon(my_uuid);
```
Comment (Keruspe) [inline src/lib.rs]: isn't the type inference enough here?

**Comment by Keruspe** [comment]
Apart from the trivial remarks, LGTM once you fix the build failure

File: src/lib.rs (hunk 2 above)
Comment (chpio) [reply src/lib.rs]: yeah, you see types often in examples so the reader can understand what's going on without looking into the specific fn definition. i can remove them if you want

Comment (chpio) [reply src/lib.rs]: removed them, in my own project i would keep them, but it's a very simple API so whatever...

Comment (Keruspe) [reply src/lib.rs]: I get it, I don't really mind the types being explicit, that was just a question.
This is simple enough though, so I don't think it changes much.

**Pull request merged**